settlement/bitflyer/upload.go

```diff
@@ -295,139 +285,154 @@ func createBitflyerRequests(
+		return nil, err
 	}
-
-	for _, bulkPayoutFile := range bulkPayoutFiles {
-		bytes, err := ioutil.ReadFile(bulkPayoutFile)
-		if err != nil {
-			logger.Error().Err(err).Msg("failed to read bulk payout file")
-			return submittedTransactions, err
+	for i, batch := range transactionBatches {
+		for j, tx := range batch {
+			tx.ProviderID = tx.BitflyerTransferID()
+			batch[j] = tx
 		}
+		transactionBatches[i] = batch
+	}
 
-		var txs []settlement.Transaction
-		err = json.Unmarshal(bytes, &txs)
-		if err != nil {
-			logger.Error().Err(err).Msg("failed unmarshal bulk payout file")
-			return submittedTransactions, err
-		}
-		for i, tx := range txs {
-			tx.ProviderID = tx.TransferID()
-			txs[i] = tx
-		}
-		// group by publisher and transfer id
-		groupedByPublisher := GroupSettlements(&txs)
-		// bat limit
-		probiLimit := altcurrency.BAT.ToProbi(decimal.NewFromFloat32(200000). // start with jpy
-											Div(quote.Rate). // convert to bat
-			// Mul(decimal.NewFromFloat(0.9)). // reduce by an extra 10% if we're paranoid
-			Truncate(8)) // truncated to satoshis
-		limitedSettlements, transactionGroups, notSubmittedSettlements, err := setupSettlementTransactions(
-			groupedByPublisher,
-			probiLimit,
-			excludeLimited,
-		)
-		if err != nil {
-			return nil, err
-		}
-		if len(notSubmittedSettlements) > 0 {
-			submittedTransactions[notSubmittedCategory] = append(
-				submittedTransactions[notSubmittedCategory],
-				notSubmittedSettlements...,
-			)
-		}
+	requests, err := createBitflyerRequests(
+		sourceFrom,
+		dryRun,
+		quote.PriceToken,
+		transactionBatches,
+	)
+	if err != nil {
+		return nil, err
+	}
 
-		limitedSettlementsByTransferID := mapSettlementsByTransferID(limitedSettlements)
+	if len(*requests) != len(transactionBatches) {
+		return nil, errors.New("number of requests doesn't match number of batches!")
+	}
 
-		requests, err := createBitflyerRequests(
-			sourceFrom,
-			dryRun,
-			quote.PriceToken,
-			transactionGroups,
-		)
-		if err != nil {
-			return nil, err
-		}
-		for i, request := range *requests {
-			if action == "upload" {
-				submittedTransactions, err = SubmitBulkPayoutTransactions(
-					ctx,
-					limitedSettlementsByTransferID,
-					groupedByPublisher,
-					submittedTransactions,
-					request,
-					bitflyerClient,
-					len(bulkPayoutFiles)-1,
-					i,
-				)
-				if err != nil {
-					logger.Error().Err(err).Msg("failed to submit bulk payout transactions")
-					return nil, err
-				}
-			} else if action == "checkstatus" {
-				submittedTransactions, err = CheckPayoutTransactionsStatus(
-					ctx,
-					limitedSettlementsByTransferID,
-					groupedByPublisher,
-					submittedTransactions,
-					request,
-					bitflyerClient,
-					len(bulkPayoutFiles)-1,
-					i,
-				)
-				if err != nil {
-					logger.Error().Err(err).Msg("falied to check payout transactions status")
-					return nil, err
-				}
+	for i, request := range *requests {
+		if action == "upload" {
```

Comment:
Instead of the string `upload` or `checkstatus`, can we just pass the function `SubmitBulkPayoutTransaction` or `CheckPayoutTransactionStatus` ? We could then simplify here to just run the command passed.